rfcs/0000-terraform-workflow-improvements.md

```diff
@@ -0,0 +1,826 @@
+1. Switch to the feature branch and run `terraform plan`/`terraform apply` to see the impact of the changes on the pre-existing resources
+1. When finished testing, run `terraform destroy` to tear down all test resources
+
+This workflow is intended to be carried out manually at first, but we might be able to use GitHub Actions to automate it at some point.
+
+#### Caveats for testing ECS applications
+
+In order to be able to test ECS applications, we will need to figure out how to allow the ECS service to retrieve a container image for deployment. There are a few blockers to this that will need to be overcome:
+
+- We'll need to override the ECR repo URL with the production AWS account's repo, and give the test account permission to pull images
+- We'll need to address an issue where apps can't start until the first GitHub Actions deploy because the template task definition references `latest`:
+  - Option 1: Add `latest` tag to all ECR images
+  - Option 2: Allow overriding image tag in module with [`.tfvars` files][terraform-tfvars].
+
+### Isolated Testing Environment
+
+As noted in the [guide-level explanation][guide-module-testing], testing will happen in the separate `mbta-ctd-test` AWS account. This account will be accessed using "assume role" permissions that full access to manage required AWS resources. The assumed role won't have full administrator access, but we'll manage the list of permitted services via IAM policies.
+
+## 3. Standardizing on Infrastructure Component Modules
+[reference-standardization]: #3-standardizing-on-infrastructure-component-modules
+
+As noted in the [guide-level explanation][guide-standardization], this is currently the most mature of the proposed improvements, in that the `ctd-ecs-app` and `ctd-rds-db` modules already exist and are seeing use. Aside from increasing adoption and carrying out the [testing workflow][guide-module-testing] and [module versioning][guide-module-versioning] recommendations, other improvements that will be beneficial include:
+
+- Handling all use cases for each module, using vars to enable/disable features that aren't likely to be used everywhere
+- Building in support for maintenance modes to minimize temporary configuration changes
+- Standardizing on the use of the `context` variable
+
+## 4. Module Versioning/Publishing
+[reference-module-versioning]: #4-module-versioningpublishing
+
+As noted in the [guide-level explanation][guide-module-versioning], module versioning and publishing is something we will set up for all infrastructure component modules, but which should be considered optional for application modules. See also the [corresponding subsection under rationale and alternatives][alternatives-module-versioning].
+
+### How To Publish A Module
+
+To set up a module for versioning and publishing, we'll need to:
+
+- Move the module's configuration to a separate repo, for a few reasons:
```

Comment:
can (should?) that separate repo be the one that contains the code that is run on that infrastructure?